LSPS1/README.md

```diff
@@ -286,48 +288,124 @@ The client MAY check the current status of the order at any point.
+    "fee_total_sat": 8888,
+    "order_total_sat": "200888",
+    "state": "EXPECT_PAYMENT"
+  },
+  "onchain": {
     "state": "EXPECT_PAYMENT",
-    "fee_total_sat": "8888",
-    "order_total_sat": "2008888",
-    "bolt11_invoice": "lnbc252u1p3aht9ysp580g4633gd2x9lc5al0wd8wx0mpn97...",
+    "fee_total_sat": 9999,
+    "order_total_sat": 200999,
     "onchain_address": "bc1p5uvtaxzkjwvey2tfy49k5vtqfpjmrgm09cvs88ezyy8h2zv7jhas9tu4yr",
     "min_onchain_payment_confirmations": 1,
     "min_fee_for_0conf": 253,
     "onchain_payment": {
-        "outpoint": "0301e0480b374b32851a9462db29dc19fe830a7f7d7a88b81612b9d42099c0ae:1",
-        "sat": "1200",
-        "confirmed": false
+      "outpoint": "0301e0480b374b32851a9462db29dc19fe830a7f7d7a88b81612b9d42099c0ae:1",
+      "sat": "1200",
+      "confirmed": false
     }
-},
+  }
+}
 ```
 
-- `state <string enum>` MUST be one of these values:
+The Client MUST ignore unkown payment options.
+
+The LSP MAY omit payment options.
+
+> **Rationale**: E.g: The LSP might not support onchain payments.
+
+> **Rationale**:
+> Fees and expiry dates are defined on the level of the payment-option.
+> This allows an LSP to provide the cheapest service for each payment option.
+>
+> An LSP might choose to have a lower fee for lightning than for onchain payments.
+> Onchain payments might
+> - be more costly for the LSP as they might have to spend a small UTXO in the future
+> - be more risky for the LSP in case of 0-conf channels
+> - require longer confirmation times than a lightning payment. (Risk of changing on-chain fees)
+
+
+#### 3.1 Lightning Payments using BOLT-11
+
+```json
+"bolt11": {
+    "state" : "EXPECT_PAYMENT",
+    "bolt11_invoice": "lnbc252u1p3aht9ysp580g4633gd2x9lc5al0wd8wx0mpn97..."
+    "fee_total_sat": 8888,
+    "order_total_sat": "200888",
+    "state": "EXPECT_PAYMENT"
+}
+```
+
+- `state`
     - `EXPECT_PAYMENT` Payment expected.
     - `HOLD` Lighting payment arrived, preimage NOT released.
-    - `PAID` Lightning payment arrived, preimage released OR full `order_total_sat` on-chain payment arrived.
-    - `REFUNDED` Lightning payment or on-chain payment has been refunded.
+    - `PAID`  When the has been reimage released
+    - `REFUNDED` Lightning payment has been refunded.
+- `expires_at` <[LSPS0.datetime][]> The timestamp at which the
 - `fee_total_sat` <[LSPS0.sat][]> The total fee the LSP will charge to open this channel in satoshi.
 - `order_total_sat` <[LSPS0.sat][]> What the client needs to pay in total to open the requested channel.
-  - MUST be the `fee_total_sat` plus the `client_balance_sat` requested in satoshi.
+    - MUST be the `fee_total_sat` plus the `client_balance_sat` requested in satoshi.
 - `bolt11_invoice <string>`
     - MUST be a [Lightning BOLT 11 invoice](https://github.com/lightning/bolts/blob/master/11-payment-encoding.md) for the number of `order_total_sat`. 
     - Invoice MUST be a [HOLD invoice](https://bitcoinops.org/en/topics/hold-invoices/).
```

Comment:
I see the point now.

I didn't change the wording and prefer to not tackle it in this MR.
I'll provide a bit of background below. But prefer to open a new MR or issue if we want to continue the discussion.

**background**
An LSP-server needs to be able to refund the customer when the channel open fails.
For on-chain payments the client provides a refund-address.
For bolt11-payments we don't need a refund-address, the server can refund the client by revoking the HTLC.

The HOLD-invoice is just a plain and normal BOLT-11 invoice.

However, instead of accepting the payment immediately, the server will hold the payment and open the channel. Once the channel is successfully opened the server will accept the payment. If the server fails to open the channel it can refund the payment by revoking the HTLC's.

In Core Lightning this behavior can be implemented using the `invoice_payment`-hook.
I encountered the word HOLD-invoice first in an LND-context. Their implementation introduces the `addholdinvoice`-command which creates a normal invoice but `lnd` will hold the payment instead of accepting it. It will only accept or refuse the payment when the user explicitly tells `lnd` to do this.